search/indexing_api_test.py

```diff
@@ -97,21 +100,42 @@ def get_default_backing_index(self):
 
 def get_sources(results):
     """get sources from es hits"""
-    return sorted([hit['_source'] for hit in results['hits']], key=lambda hit: hit['id'])
+    sorted_hits = sorted(results['hits'], key=lambda hit: hit['_source']['id'])
+    return (
+        [hit['_source'] for hit in sorted_hits if hit['_type'] == USER_DOC_TYPE],
+        [hit['_source'] for hit in sorted_hits if hit['_type'] == PUBLIC_USER_DOC_TYPE],
+    )
+
+
+def remove_es_keys(hit):
+    """
+    Removes ES keys from a hit object
```

Comment:
Can you add arguments and a return value for this, and also a warning saying that `hit` will be modified?